### PR TITLE
Update typography docs and Storybook theming

### PR DESCRIPTION
**Pull request by VincentSmedinga**
This structures our typography documentation:

- Reorders and groups the sections
- Uses the [`Typeset` doc block](https://storybook.js.org/docs/react/api/doc-block-typeset) – with static font sizes or it will render long `clamp` labels
- Reformulates some of the guidelines
- Removes the maximum sentence length of 80 characters
- Mentions the font license and form

It also improves readability of our docs by enlarging the tiny prose text, applying our own font and increasing whitespace. There appears to be no better way to achieve this than to bluntly override CSS (see [issues](https://github.com/storybookjs/storybook/issues?q=is%3Aissue+theming+font+size+is%3Aopen)). Also `textSize` in `themeConfig` has no effect. Suggestions welcome, I do think we need the end result.

### CODE REVIEW

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 1]
```diff
@@ -94,11 +170,27 @@ Een manier om dat te doen is met de volgende _css snippet_, waarin `hyphens` en
...
-## Hoe te gebruiken?
+## Het lettertype installeren
+
+Als je de React-componenten gebruikt hoef je niets te doen.
+Deze gebruiken allemaal het juiste lettertype.
+
+In andere gevallen installeer je Amsterdam Sans op 1 van deze 2 manieren:
+
+1. Kopieer [deze regels CSS](https://amsterdam.nl/publish/pages/1007152/amsterdam_sans_v_0_9.css) naar je applicatie.
+   Het maakt gebruik van ons snelle CDN.
```
Comment (dlnr) [inline storybook/storybook-docs/src/typography.stories.mdx:181]: Fast CDN? They are uploads to amsterdam.nl, which will be changing pretty soon. If we use an actual CDN with proper headers for the font files we can change the imports in the package and link this to the [npm code view](https://www.npmjs.com/package/@amsterdam/design-system-assets?activeTab=code). Or just use a code block in this document.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 2]
```diff
@@ -1,84 +1,160 @@
...
+
+De meeste tekst in het design system is puur zwart op puur wit en dat voldoet daar ruim aan.
+Op de meeste donkere achtergronden maken we tekst wit; ook dat biedt voldoende contrast.
+Voor sommige achtergrondkleuren is ook zwarte tekst goed leesbaar.
+Raadpleeg de betreffende componenten voor specifieke richtlijnen.
 
-De Nederlandse taal kent veel lange woorden, zoals 'gemeenteraadsverkiezingen'.
+### Uitlijning
+
+Tekst wordt nagenoeg altijd links uitgelijnd.
```
Comment (dlnr) [inline storybook/storybook-docs/src/typography.stories.mdx:148]: I silently hope we'll be working on a 'right to left' or even vertical version for translated pages.

**Review by dlnr — COMMENTED** [review]
Documentation looks great with the TypeSet component, great find!

File: storybook/storybook-docs/src/grid.stories.mdx
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/storybook-docs/src/grid.stories.mdx:1]: > Documentation looks great with the TypeSet component, great find!
Thanks! Wait until you see the Color and Icon docs I’ve planned! 🤩

> Did I change too soon?
Nope, no problem. I’ll translate everything we have in one go, feel free to be early. For this doc, the aim was restructuring, not translating.

File: proprietary/assets/font/index.css [hunk 3]
```diff
@@ -15,6 +15,7 @@
     url("./woff2/AmsterdamSans-Light.woff2") format("woff2"),
     url("./woff/AmsterdamSans-Light.woff") format("woff");
 }
+
 @font-face {
   font-display: swap;
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply proprietary/assets/font/index.css:20]: Sorry I only added whitespace here. Can you create a ticket to discuss font loading and resolve this one?

File: storybook/storybook-docs/config/preview-head.html
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/storybook-docs/config/preview-head.html:1]: Yes, that would be better. I’ll try and change that.

File: storybook/storybook-docs/src/typography.stories.mdx (hunk 2 above)
Comment (VincentSmedinga) [reply storybook/storybook-docs/src/typography.stories.mdx:148]: Sure we will! At least the right-to-left one, not sure about top-to-bottom 🥲

File: storybook/storybook-docs/src/typography.stories.mdx (hunk 1 above)
Comment (VincentSmedinga) [reply storybook/storybook-docs/src/typography.stories.mdx:181]: No idea, I copied this from the old site. Should have opened the link :)

What do you say, remove it and just suggest going with our package?

File: storybook/storybook-docs/config/preview-head.html
[the remark this answers is not in the document]
Comment (dlnr) [reply storybook/storybook-docs/config/preview-head.html:1]: Sorry what I was referring to was this converstation: https://github.com/Amsterdam/design-system/pull/681/commits/654259a2e6db569147fdd492cd876299f50d40a4#r1389541766, so to put the css from the html file into a css file and importing it to preview.tsx

File: proprietary/assets/font/index.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (dlnr) [reply proprietary/assets/font/index.css:20]: Yes should be fine, an actual CDN should be on the radar though.

File: storybook/storybook-docs/src/typography.stories.mdx (hunk 1 above)
Comment (dlnr) [reply storybook/storybook-docs/src/typography.stories.mdx:181]: Not sure if downloading our font is following the license policy, from a CDN we would have some form of control, but I guess you can never really stop anyone from downloading/converting the font.

File: storybook/storybook-docs/config/preview-head.html
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/storybook-docs/config/preview-head.html:1]: That link doesn’t show a comment for me.

I am including the CSS in the TypeScript file now, right? https://github.com/Amsterdam/design-system/pull/771/files#diff-36f3806e40237d121902759b9425ac15fb1f35b30ba74db64ed037b3db31a3f6R3

[the remark this answers is not in the document]
Comment (dlnr) [reply storybook/storybook-docs/config/preview-head.html:1]: Right, I'll add the lines for the dialog manually after this is merged.

File: storybook/storybook-docs/src/typography.stories.mdx (hunk 1 above)
Comment (VincentSmedinga) [reply storybook/storybook-docs/src/typography.stories.mdx]: I think so too. I’ve updated the text.

**Review by dlnr — APPROVED** [review]

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 4]
```diff
@@ -1,84 +1,160 @@
...
 
 <Meta title="Docs/Designrichtlijnen/Typografie" />
 
 # Typografie
 
 ## Lettertype
 
-We gebruiken het lettertype Amsterdam Sans.
-De _fallback_ lettertypes zijn Arial en sans-serif.
+De gemeente gebruikt de Amsterdam Sans als beeldbepalend lettertype.
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:9]: This change differs a bit from the [Schrijfwijzer guidelines](https://www.amsterdam.nl/schrijfwijzer/heldere-taal-basis-onze-huisstijl/). They use a lot of 'we' there, not 'de gemeente'. Imo, the original sentence was more simple and direct.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 5]
```diff
@@ -1,84 +1,160 @@
...
 
-We gebruiken 2 varianten:
+Dat is een letterfamilie met een helder, open tekstbeeld dat goed bij de Andreaskruisen past.
+Het lettertype is zowel offline als online goed leesbaar.
+Daarom wordt het in alle communicatiemiddelen ingezet.
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:13]: We gebruiken het in alle communicatiemiddelen.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 6]
```diff
@@ -1,84 +1,160 @@
...
 
-- Amsterdam Sans Regular
-- Amsterdam Sans ExtraBold
+Deze zin demonstreert alle letters van het Latijnse alfabet en cijfers in de Amsterdam Sans:
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:15]: 'de Amsterdam Sans' feels a bit weird to me, why not just 'Amsterdam Sans'?

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 7]
```diff
@@ -1,84 +1,160 @@
...
-| ----: | :----------------------------------- |
-|     0 | Page heading                         |
-|     1 | Heading 1                            |
-|     2 | Heading 2                            |
-|     3 | Heading 3, Blockquote                |
-|     4 | Heading 4, Top Task Link             |
-|     5 | Paragraph large                      |
-|     6 | Paragraph, alle Links behalve inList |
-|     7 | Paragraph small, inList Link         |
+| Componenten                                          | Niveau |
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:63]: First 'Niveau' and then 'Componenten' was more readable imo, because we're talking about the text levels in this section.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 8]
```diff
@@ -1,84 +1,160 @@
...
+### Stijl
 
-- Op knoppen worden labels gecentreerd.
-- In tabellen kan de inhoud van kolommen, afhankelijk van de inhoud, ook gecentreerd of rechts uitgelijnd worden.
+Gebruik de cursieve weergave uitsluitend om fragmenten in tekst te accentueren.
+Cursieve tekst is minder goed leesbaar dan reguliere of vetgedrukte tekst.
 
-## Vormgeving
+Pas het niet toe op hele tekstblokken.
+Er zijn betere manieren om lange teksten nadruk te geven.
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:105]: Imo, this is also less simple than the original text. How about:

Vermijd cursieve teksten  zo veel mogelijk, deze zijn niet goed te lezen.
Gebruik ze uitsluitend om tekst te accentueren, niet voor hele tekstblokken.
Er zijn betere manieren om lange teksten nadruk te geven.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 9]
```diff
@@ -1,84 +1,160 @@
...
+  fontSizes={[24]}
+  sampleText="Jouw typograaf biedt mij zulke exquise schreven"
+  style={{ fontStyle: "italic" }}
+/>
 
-Vermijd het onderlijnen van tekst.
-Reserveer het onderlijnen van tekst voor interactieve links.
+### Onderstreping
+
+Uitsluitend de tekst van een hyperlink wordt onderstreept.
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:116]: Same here. How about:

Vermijd het onderstrepen van tekst. Onderstreep alleen interactieve links.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 10]
```diff
@@ -1,84 +1,160 @@
...
+Rond staartletters wordt de lijn onderbroken voor betere leesbaarheid.
+
+Onderstreepte tekst zonder hyperlink geeft toch de suggestie van een link.
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:119]: Onderstreepte teksten lijken op links.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 11]
```diff
@@ -1,84 +1,160 @@
...
+## Accentuering
 
-Het contrast tussen tekst en achtergrondkleur moet voldoen aan [de laatste wettelijke toegankelijkheidseisen van de WCAG-niveau AA](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).
+Er zijn verschillende manieren om tekst te accentueren.
+Hierin hebben we de volgende keuzes gemaakt.
 
-## Uitlijning
+### Gewicht
 
-Het aantal karakters op een regel is maximaal 80.
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:59]: Why did you remove this? It's kind of an important rule, especially with our huge font sizes.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 12]
```diff
@@ -1,84 +1,160 @@
...
+
+### Contrast
+
+Het contrast tussen tekst en achtergrondkleur moet voldoen aan de laatste wettelijke toegankelijkheidseisen van [de WCAG, niveau AA](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).
 
-De Nederlandse taal kent veel lange woorden, zoals 'gemeenteraadsverkiezingen'.
+De meeste tekst in het design system is puur zwart op puur wit en dat voldoet daar ruim aan.
+Op de meeste donkere achtergronden maken we tekst wit; ook dat biedt voldoende contrast.
+Voor sommige achtergrondkleuren is ook zwarte tekst goed leesbaar.
+Raadpleeg de betreffende componenten voor specifieke richtlijnen.
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:144]: This is a bit vague. Shouldn't we [link to Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2%3A294&mode=dev) here? There's an overview of what text color goes with what background color there.

File: storybook/storybook-docs/src/typography.stories.mdx [hunk 13]
```diff
@@ -94,11 +170,24 @@ Een manier om dat te doen is met de volgende _css snippet_, waarin `hyphens` en
 
 Voor verdere informatie: [Wrapping and breaking text op MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_text/Wrapping_breaking_text)
 
-## Hoe te gebruiken?
+## Het lettertype installeren
+
+Als je de React-componenten gebruikt hoef je niets te doen.
+Deze gebruiken allemaal het juiste lettertype.
```
Comment (alimpens) [inline storybook/storybook-docs/src/typography.stories.mdx:176]: This is currently not exactly true. If you want to use Amsterdam Sans, you have to install it seperately.

Comment (VincentSmedinga) [reply storybook/storybook-docs/src/typography.stories.mdx:176]: Yes, nuanced this.

**Review by VincentSmedinga — COMMENTED** [review]
@alimpens I’ve applied your feedback to #803.

File: storybook/storybook-docs/src/typography.stories.mdx (hunk 11 above)
Comment (alimpens) [reply storybook/storybook-docs/src/typography.stories.mdx:59]: True, but in practice the huge font sizes probably mean you should have less than 80 characters on a line. We ran into this issue regularly with onderzoek.amsterdam.nl, the large Paragraph font size spanning 10 columns or so is hard to read, you have to swivel your head from left to right to much. 

I'd really like to test the readability of the different max chars on a line for the different font sizes we have. But until that time, we should at least set a maximum width imo. We could define a maximum amount of columns or something, but I think a maximum amount of chars is easier to work with.
